File: 3909.compare-sums-of-bitonic-parts/solution.go
func compareBitonicSums(nums []int) int {
    n := len(nums)
    sumA := nums[0]
    i := 1

    for i < n && nums[i] > nums[i-1] {
        sumA += nums[i]
        i++
    }

    sumD := nums[i-1]
    for i < n {
        sumD += nums[i]
        i++
    }

    if sumA > sumD {
        return 0
    }
    if sumA < sumD {
        return 1
    }
    return -1
}

File: 3909.compare-sums-of-bitonic-parts/solution.kt
class Solution {
    fun compareBitonicSums(nums: IntArray): Int {
        var sum = nums[0].toLong()
        var i = 0

        while (i < nums.lastIndex && nums[i + 1] > nums[i]) {
            sum += nums[++i]
        }

        while (i <= nums.lastIndex) {
            sum -= nums[i]
            i++
        }

        return when {
            sum > 0 -> 0
            sum < 0 -> 1
            else -> -1
        }
    }
}